content/spin/http-trigger.md

```diff
@@ -35,54 +44,72 @@ by defining the top-level application trigger:
 
-- an HTTP component configured on the `/goodbye` route that uses the Wagi executor:
+A _wildcard_ route matches the given route and any route under it.  A route is a wildcard if it ends in `/...`.  For example, `/users/...` matches `/users`, `/users/1`, `/users/1/edit`, and so on.  Any of these routes will run the mapped component.
+
+> In particular, the route `/...` matches all routes.
 
 <!-- @nocpy -->
 
 ```toml
+[[component]]
+id = "user-manager"
+source = "users.wasm"
+# Run the `users.wasm` module when the application receives a request to `/users`
+# or any path beginning with `/users/`
 [component.trigger]
-route = "/goodbye"
-executor = { type = "wagi" }
+route = "/users/..."
 ```
 
-## Routing
+### Routing with an Application `base`
+
+If the application `base` is `"/"` then all component routes are matched exactly as given.
 
-Routing an incoming request to a particular component is done using the
-application base path (`base` in `spin.toml`) and the component defined routes
-(`route` in the component configuration) by prefixing the application base path
-to all component routes defined for that application.
+If `base` contains a non-root path, this is prefixed to all component routes,
+exact or wildcard.
 
-For example, if the application `base` path is `base = /base`, and a component
-has defined `route = /foo`, that component will be executed for requests on
-`http(s)://<spin-up-defined-address-and-port>/base/foo`.
+For example, suppose the application `base` path is `base = "/shop"`.  Then a component with `route = "/cart"` will be executed for requests to `/shop/cart`.  Similarly, a component with `route = "/users/..."` will be executed for requests to `/shop/users`, `/shop/users/1`, `/shop/users/1/edit` and so on.
 
-Components can either define exact routes, for example `route = /bar/baz`, where
-the component will be invoked only for requests on `/base/bar/baz`, or they
-can define a wildcard as the last path segment, for example `route = /bar/baz/...`,
-which means the component will be invoked for every request starting with the
-`/base/bar/baz/` prefix (such as `/base/bar/baz`, `/base/bar/baz/qux`,
-`/base/bar/baz/qux/quux` and so on).
+### Resolving Overlapping Routes
 
 If multiple components could potentially handle the same request based on their
 defined routes, the component whose route has the longest matching prefix 
-takes precedence. 
+takes precedence.  This also means that exact matches take precedence over wildcard matches.
 
-In the following example, any request starting with the  `/foo/` prefix (e.g. `/foo/bar`)
-will be handled by `component-1`:
+In the following example, requests starting with the  `/users/` prefix (e.g. `/users/1`)
+will be handled by `user-manager`, even though it is also matched by the `shop` route, because the `/users` prefix is longer than `/`.
+But requests to `/users/admin` will be handled by the `admin` component, not `user-manager`, because that is a more exact match still.
```

Comment:
```suggestion
But requests to `/users/admin` will be handled by the `admin` component, not `user-manager`, because that is a more exact match still:
```